### ISSUE TITLE
helm dependency build fails when repo URL contains escaped 

### ISSUE THREAD

**Issue by sathieu** [helm/helm#9821]
With the following : `Chart.yaml`:

```yaml
apiVersion: v2
name: "gatekeeper"
version: 0.0.1
dependencies:
  - name: gatekeeper
    repository: https://gitlab.example.org/api/v4/projects/external-packages%2fgatekeeper/packages/helm/stable/
    version: 3.4.0
```

(the ending `/` is to workaround #9764).

`helm dependency build` failed exit status 1:

```
Error: could not download https://gitlab.example.org/api/v4/projects/external-packages/gatekeeper/packages/helm/stable/charts/gatekeeper-3.4.0.tgz:
failed to fetch https://gitlab.example.org/api/v4/projects/external-packages/gatekeeper/packages/helm/stable/charts/gatekeeper-3.4.0.tgz
```

As you can see, `%2` has been replaced by `/`.

**Comment by hickeyma** [comment]
@sathieu I tried to reproduce the issue using the latest code from `main` and it looks like it is now working. Here is how I tested:

```console
$ cat pr-9822/Chart.yaml

apiVersion: v2
name: pr-9822
description: A Helm chart for Kubernetes

[...]

dependencies:
  - name: kubitus-ingress-operator
    repository: https://gitlab.com/api/v4/projects/kubitus-project%2Fkubitus-ingress-operator/packages/helm/main
    version: 0.1.0-20210728154159-bdb9fe13

 $ helm dep build pr-9822

Getting updates for unmanaged Helm repositories...
...Successfully got an update from the "https://gitlab.com/api/v4/projects/kubitus-project%2Fkubitus-ingress-operator/packages/helm/main" chart repository
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "prometheus-community" chart repository
...Successfully got an update from the "iks-charts" chart repository
...Successfully got an update from the "bitnami" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading kubitus-ingress-operator from repo https://gitlab.com/api/v4/projects/kubitus-project%2Fkubitus-ingress-operator/packages/helm/main
Deleting outdated charts
```

A fix may have been applied in the meantime?

**Comment by sathieu** [comment]
@hickeyma Yes, probably fixed by another commit from me (d506314abfb5d21419df8c7e7e68012379db2354), which is in v3.6.3.

unfortunately, ArgoCD has not updated it yet (https://github.com/argoproj/argo-cd/pull/6654).

**Comment by hickeyma** [comment]
@sathieu Do you mind testing this against v3.6.3 and making sure it is working as expected? If it is then we should close this issue as already fixed.
